crates/scroll/rpc/src/error.rs

```diff
@@ -0,0 +1,100 @@
+//! RPC errors specific to Scroll.
+
+use alloy_rpc_types_eth::BlockError;
+use jsonrpsee_types::error::INTERNAL_ERROR_CODE;
+use reth_rpc_eth_api::AsEthApiError;
+use reth_rpc_eth_types::EthApiError;
+use reth_rpc_server_types::result::internal_rpc_err;
+use revm::primitives::InvalidTransaction;
+
+use reth_scroll_evm::ScrollBlockExecutionError;
+
+/// Scroll specific errors, that extend [`EthApiError`].
+#[derive(Debug, thiserror::Error)]
+pub enum ScrollEthApiError {
+    /// L1 ethereum error.
+    #[error(transparent)]
+    Eth(#[from] EthApiError),
+    /// EVM error originating from invalid Scroll data.
+    #[error(transparent)]
+    Evm(#[from] ScrollBlockExecutionError),
+    /// Thrown when calculating L1 gas fee.
+    #[error("failed to calculate l1 gas fee")]
+    L1BlockFeeError,
+    /// Thrown when calculating L1 gas used
+    #[error("failed to calculate l1 gas used")]
+    L1BlockGasError,
+    /// Wrapper for [`revm_primitives::InvalidTransaction`](InvalidTransaction).
+    #[error(transparent)]
+    InvalidTransaction(#[from] ScrollInvalidTransactionError),
+    /// Sequencer client error.
+    #[error(transparent)]
+    Sequencer(#[from] SequencerClientError),
```

Comment:
Do we ever use any of these error types? I had a look and I can't find any references to them.

Reply:
will clean the unused error